src/Files.App/ViewModels/ToolbarViewModel.cs

```diff
@@ -1150,9 +1150,11 @@ public List<ListedItem> SelectedItems
 		public bool CanEmptyRecycleBin => InstanceViewModel.IsPageTypeRecycleBin && HasItem;
 		public bool CanRestoreRecycleBin => InstanceViewModel.IsPageTypeRecycleBin && HasItem && (SelectedItems is null || SelectedItems.Count == 0);
 		public bool CanRestoreSelectionRecycleBin => InstanceViewModel.IsPageTypeRecycleBin && HasItem && SelectedItems is not null && SelectedItems.Count > 0;
-		public bool CanExtract => SelectedItems is not null && SelectedItems.Any() && SelectedItems.All(x => FileExtensionHelpers.IsZipFile(x.FileExtension)) && !InstanceViewModel.IsPageTypeRecycleBin;
-		public string ExtractToText => CanExtract ? SelectedItems.Count > 1 ? string.Format("ExtractToChildFolder".GetLocalizedResource(), $"*{Path.DirectorySeparatorChar}") : string.Format("ExtractToChildFolder".GetLocalizedResource() + "\\", Path.GetFileNameWithoutExtension(selectedItems.First().Name)) : "ExtractToChildFolder".GetLocalizedResource();
-		public bool IsMultipleArchivesSelected => CanExtract && SelectedItems.Count > 1;
+		public bool CanExtract => IsArchiveOpened ? (SelectedItems is null || !SelectedItems.Any()) : IsArchiveSelected;
+		public string ExtractToText => IsArchiveSelected ? SelectedItems.Count > 1 ? string.Format("ExtractToChildFolder".GetLocalizedResource(), $"*{Path.DirectorySeparatorChar}") : string.Format("ExtractToChildFolder".GetLocalizedResource() + "\\", Path.GetFileNameWithoutExtension(selectedItems.First().Name)) : "ExtractToChildFolder".GetLocalizedResource();
+		public bool IsArchiveOpened => FileExtensionHelpers.IsZipFile(Path.GetExtension(pathControlDisplayText));
+		public bool IsArchiveSelected => SelectedItems is not null && SelectedItems.Any() && SelectedItems.All(x => FileExtensionHelpers.IsZipFile(x.FileExtension)) && !InstanceViewModel.IsPageTypeRecycleBin;
+		public bool IsMultipleArchivesSelected => IsArchiveSelected && SelectedItems.Count > 1;
```

Comment:
Shouldn't it check if all the selected items are archives as well?

Reply:
which one?

Reply:
Sorry, still a bit new on reviews, seems I selected more lines than I intended too.

I meant `IsMultipleArchivesSelected`, shouldn't it check that all the items selected are only archives?

Reply:
If you mean that `IsMultipleArchivesSelected` should check if all items are archives, that is already checked by `IsArchiveSelected`

Reply:
Alright, so we got it covered. Wouldn't be the place I expect to such a test, but it works.

Reply:
Actually I didn't want to name it `IsArchiveSelected`, but I couldn't find any other more suitable name.

Reply:
I'd propose to rename it `HasSelectionOneArchiveOnly` and then `IsMultipleArchivesSelected` into `HasSelectionMultipleArchivesOnly`.

Then extract `SelectedItems is not null && SelectedItems.Any() && SelectedItems.All(x => FileExtensionHelpers.IsZipFile(x.FileExtension))  && !InstanceViewModel.IsPageTypeRecycleBin` into a method named `IsSelectionArchivesOnly`.

You will end up with :

`public bool HasSelectionOneArchiveOnly => IsSelectionArchivesOnly && SelectedItems.Count == 1`
and 
`public bool HasSelectionMultipleArchivesOnly => IsSelectionArchivesOnly && SelectedItems.Count > 1`